libweb3jsonrpc/UnixSocketServer.cpp

```diff
@@ -103,11 +104,27 @@ bool UnixDomainSocketServer::StopListening()
 	return false;
 }
 
+static bool waitForReadable(int fd, unsigned timeoutMillis)
+{
+	fd_set in, out, err;
+	FD_ZERO(&in);
+	FD_ZERO(&out);
+	FD_ZERO(&err);
+	FD_SET(fd, &in);
+	struct timeval tv;
```

Comment:
In a more C++ way this could be `timeval const tv = {  timeoutMillis / 1000, timeoutMillis % 1000 }`